### PR TITLE
Global nav role switcher hooked up

### PR DESCRIPTION
**Pull request by colbycheeze**
Addresses story #2 and task #5

Still need to:
 - Add hover css to role switcher
 - fix some padding
 - Make sure it is keyboard accessible
 - Add a "create new user" UI component (will need another story for that, since it has to have some UI to help you select a state to associate with that user)
 - Remove all "font" icons, and replace with the SVG icons (I did ONE of them for now as an example)
 - Fix the story for GlobalNav (it's missing the redux context, which makes it fail...we can export the state from Redux dev tools in order to set up a store context)

Hooking this up took a bit of refactoring.

The dashboard page currently doesn't call the getAdminData. I'll hook that up next to a real lifecycle or action.

Here is what everything looks like:
screen recording 2016-09-08 at 04 48 pm

## Other changes in this PR
 - Added an ENV var pointing to the dev api for npm start (we can remove it at a later time when we expect public users to begin using the demo, but it makes things easy for us atm.)
 - Tests now inject the tap event plugin, so that MUI components dont spit out errors
 - Test config removes extra quotes from stringified json strings.

### Demo Data Flow
 - demo module now handles fetching data associated with a demo session and logs in to the supplychainmanager automatically
 - Dashboard component no longer manages data fetching on route loading, instead it is handled properly within the route itself.
 - CreateDemo saga now simply creates a demo and changes the route (In a future task, I want to have it manage adding one retail manager automatically so that demos always begin with that role as a default)

### Mock API / stronger saga tests
 - Sagas tests now make use of state changes from actually calling the reducer, which requires proper data passed in....meaning I needed a way to mock the API calls. I made a mock file that gives back faker data in the shape we currently expect from the API.

What this allows, is for saga tests to properly fail if things change in the way data is passed around and modified (whereas before, we just generated made up objects and so it worked in test, but not in the running app)

The Saga tests look a bit more confusing, but are WAY more solid now. Writing and testing complex async actions has never been fun, but this makes them accurate and stable, which is the real win.

### CODE REVIEW

File: config/index.js [hunk 1]
```diff
@@ -83,9 +84,7 @@ config.globals = {
   '__DEBUG__'    : config.env === 'development' && !argv.no_debug,
   '__COVERAGE__' : !argv.watch && config.env === 'test',
   '__BASENAME__' : JSON.stringify(process.env.BASENAME || ''),
-  '__CONTROLLER_API__' : config.env === 'test' ?
-    `${config.controller_service}/api/v1` :
-    JSON.stringify(`${config.controller_service}/api/v1`),
+  '__CONTROLLER_API__' : JSON.stringify(config.controller_service || ''),
```
Comment (l2fprod) [inline config/index.js]: the removal of /api/v1 here means we will need to update the toolchain and the live pipeline. It currently only passes the host.domain part. Goal was to be able to have multiple API version and let the client - in this case the webui the choice to use v1 or v2, etc. If you want to stick to full host.domain/path make sure to update logistics-wizard-toolchain and the live pipeline.

File: package.json [hunk 2]
```diff
@@ -62,14 +62,16 @@
     "start": {
       "command": "babel-node bin/server",
       "env": {
-        "DEBUG": "app:*"
+        "DEBUG": "app:*",
+        "CONTROLLER_SERVICE": "https://dev-logistics-wizard-controller.mybluemix.net/api/v1"
```
Comment (l2fprod) [inline package.json]: this points to our live demo. In our chat yesterday we said this should not be the case but rather point to 0.0.0.0:5000 which is the local configuration defined in https://github.com/IBM-Bluemix/logistics-wizard/wiki/Quick-Start

File: config/index.js (hunk 1 above)
Comment (colbycheeze) [inline config/index.js]: Ah right. Let me update it.

**Comment by rvennam** [comment]
Learned a lot, changes look good to me - no comments.

**Pull request merged**